src/ansible_navigator/image_manager/puller.py

```diff
@@ -142,18 +151,33 @@ def prologue_stdout(self):
         print("Updating the execution environment")
         print("\u002d" * width)
 
+    def _generate_pull_command(self) -> List:
+        """Generate the pull command.
+
+        :returns: The list of command parts
+        """
+        command_line = [self._container_engine, "pull"]
+        if isinstance(self._arguments, list):
+            for argument in self._arguments:
+                command_line.extend(shlex.split(argument))
+        command_line.append(self._image)
+        return command_line
+
     def pull_stdout(self):
         """pull the image, print to stdout
 
         podman writes to stderr
         docker writes to stdout
         """
         try:
+            command_line = self._generate_pull_command()
+            shlex_joined = " ".join(shlex.quote(arg) for arg in command_line)
+            print(f"Running the command: {shlex_joined}")
             if self._container_engine == "podman":
-                subprocess.run([self._container_engine, "pull", self._image], check=True)
+                subprocess.run(command_line, check=True)
             elif self._container_engine == "docker":
                 subprocess.run(
-                    [self._container_engine, "pull", self._image],
+                    command_line,
                     check=True,
                     stderr=subprocess.PIPE,
                 )
```

Comment:
Maybe collapse this whole block into

```python
stderr_pipe = subprocess.PIPE if self.container_engine == "docker" else None
subprocess.run(command_line, check=True, stderr=stderr_pipe)
```

to reduce some duplication. It took me a minute to realize that stderr is used here only for exception processing below. That's kind of clever/evil - because at first glance, it looks like we're discarding the result of `subprocess.run()` (and therefore wouldn't care about the stderr output).

Reply:
yeah, it looks like stderr is passed to popen which has a default of None, cleaner I like.

This subtle difference between podman and docker was tricky I recall with podman not using stderr.